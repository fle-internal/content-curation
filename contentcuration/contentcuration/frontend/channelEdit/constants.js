import { fileErrors } from 'shared/constants';

export const RouterNames = {
  SANDBOX: 'SANDBOX',
  TREE_ROOT_VIEW: 'TREE_ROOT_VIEW',
  TREE_VIEW: 'TREE_VIEW',
  STAGING_TREE_VIEW: 'STAGING_TREE_VIEW',
  CONTENTNODE_DETAILS: 'CONTENTNODE_DETAILS',
  MULTI_CONTENTNODE_DETAILS: 'MULTI_CONTENTNODE_DETAILS',
  IMPORT_FROM_CHANNELS: 'IMPORT_FROM_CHANNELS',
  IMPORT_FROM_CHANNELS_BROWSE: 'IMPORT_FROM_CHANNELS_BROWSE',
  IMPORT_FROM_CHANNELS_SEARCH: 'IMPORT_FROM_CHANNELS_SEARCH',
  IMPORT_FROM_CHANNELS_REVIEW: 'IMPORT_FROM_CHANNELS_REVIEW',
  ADD_TOPICS: 'ADD_TOPICS',
  ADD_EXERCISE: 'ADD_EXERCISE',
  UPLOAD_FILES: 'UPLOAD_FILES',
  TRASH: 'TRASH',
  ADD_PREVIOUS_STEPS: 'ADD_PREVIOUS_STEPS',
  ADD_NEXT_STEPS: 'ADD_NEXT_STEPS',
};

export const viewModes = {
  DEFAULT: 'DEFAULT_VIEW',
  COMFORTABLE: 'COMFORTABLE_VIEW',
  COMPACT: 'COMPACT_VIEW',
};

export const ValidationErrors = {
  TITLE_REQUIRED: 'TITLE_REQUIRED',
  LICENCE_REQUIRED: 'LICENCE_REQUIRED',
  COPYRIGHT_HOLDER_REQUIRED: 'COPYRIGHT_HOLDER_REQUIRED',
  LICENCE_DESCRIPTION_REQUIRED: 'LICENCE_DESCRIPTION_REQUIRED',
  MASTERY_MODEL_REQUIRED: 'MASTERY_MODEL_REQUIRED',
  MASTERY_MODEL_INVALID: 'MASTERY_MODEL_INVALID',
  QUESTION_REQUIRED: 'QUESTION_REQUIRED',
  INVALID_NUMBER_OF_CORRECT_ANSWERS: 'INVALID_NUMBER_OF_CORRECT_ANSWERS',
  NO_VALID_PRIMARY_FILES: 'NO_VALID_PRIMARY_FILES',
  ...fileErrors,
};

// should correspond to backend types
export const AssessmentItemTypes = {
  SINGLE_SELECTION: 'single_selection',
  MULTIPLE_SELECTION: 'multiple_selection',
  TRUE_FALSE: 'true_false',
  INPUT_QUESTION: 'input_question',
};

export const AssessmentItemToolbarActions = {
  EDIT_ITEM: 'EDIT_ITEM',
  MOVE_ITEM_UP: 'MOVE_ITEM_UP',
  MOVE_ITEM_DOWN: 'MOVE_ITEM_DOWN',
  DELETE_ITEM: 'DELETE_ITEM',
  ADD_ITEM_ABOVE: 'ADD_ITEM_ABOVE',
  ADD_ITEM_BELOW: 'ADD_ITEM_BELOW',
};

export const AssessmentItemTypeLabels = {
  [AssessmentItemTypes.SINGLE_SELECTION]: 'questionTypeSingleSelection',
  [AssessmentItemTypes.MULTIPLE_SELECTION]: 'questionTypeMultipleSelection',
  [AssessmentItemTypes.TRUE_FALSE]: 'questionTypeTrueFalse',
  [AssessmentItemTypes.INPUT_QUESTION]: 'questionTypeInput',
};

export const TabNames = {
  DETAILS: 'details',
  PREVIEW: 'preview',
  QUESTIONS: 'questions',
  RELATED: 'related',
};

export const modes = {
  EDIT: 'EDIT',
  NEW_TOPIC: 'NEW_TOPIC',
  NEW_EXERCISE: 'NEW_EXERCISE',
  UPLOAD: 'UPLOAD',
  VIEW_ONLY: 'VIEW_ONLY',
};
