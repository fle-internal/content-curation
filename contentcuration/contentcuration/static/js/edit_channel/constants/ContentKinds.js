module.exports = [
  {
    kind: 'document',
  },
  {
    kind: 'topic',
  },
  {
    kind: 'video',
  },
  {
    kind: 'html5',
  },
  {
    kind: 'audio',
  },
  {
    kind: 'exercise',
  },
];
