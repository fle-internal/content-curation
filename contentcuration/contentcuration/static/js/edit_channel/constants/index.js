const ContentKinds = require('./ContentKinds');
const FormatPresets = require('./FormatPresets');
const Languages = require('./Languages');
const Licenses = require('./Licenses');
const MasteryModels = require('./MasteryModels');
const Roles = require('./Roles');

module.exports = {
  ContentKinds,
  FormatPresets,
  Languages,
  Licenses,
  MasteryModels,
  Roles,
};
