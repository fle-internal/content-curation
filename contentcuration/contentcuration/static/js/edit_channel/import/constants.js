export const PageTypes = {
  TREE_VIEW: 'TREE_VIEW',
  IMPORT_PREVIEW: 'IMPORT_PREVIEW',
  SEARCH_RESULTS: 'SEARCH_RESULTS',
};
