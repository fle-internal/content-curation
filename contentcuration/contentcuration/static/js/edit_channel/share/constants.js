export const Permissions = {
  OWNER: 'manage',
  EDIT: 'edit',
  VIEW_ONLY: 'view'
};

export const PermissionRanks = [
	{
		"shareMode": Permissions.VIEW_ONLY,
		"rank": 0,
		"field": 'viewers'
	},
	{
		"shareMode": Permissions.EDIT,
		"rank": 1,
		"field": 'editors'
	},
	{
		"shareMode": Permissions.OWNER,
		"rank": 2,
		"field": null
	},
]
