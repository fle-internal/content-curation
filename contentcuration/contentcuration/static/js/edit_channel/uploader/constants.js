export const TabNames = {
  DETAILS: 'DETAILS',
  PREVIEW: 'PREVIEW',
  QUESTIONS: 'QUESTIONS',
  PREREQUISITES: 'PREREQUISITES',
};

export const modes = {
  EDIT: 'EDIT',
  NEW_TOPIC: 'NEW_TOPIC',
  NEW_EXERCISE: 'NEW_EXERCISE',
  UPLOAD: 'UPLOAD',
  VIEW_ONLY: 'VIEW_ONLY',
};

export const AssessmentItemToolbarActions = {
  MOVE_ITEM_UP: 'MOVE_ITEM_UP',
  MOVE_ITEM_DOWN: 'MOVE_ITEM_DOWN',
  EDIT_ITEM: 'EDIT_ITEM',
  ADD_ITEM_ABOVE: 'ADD_ITEM_ABOVE',
  ADD_ITEM_BELOW: 'ADD_ITEM_BELOW',
  DELETE_ITEM: 'DELETE_ITEM',
};

export const AssessmentItemValidationErrors = {
  BLANK_QUESTION: 'BLANK_QUESTION',
  INVALID_NUMBER_OF_CORRECT_ANSWERS: 'INVALID_NUMBER_OF_CORRECT_ANSWERS',
};

// should correspond to backend types
export const AssessmentItemTypes = {
  SINGLE_SELECTION: 'single_selection',
  MULTIPLE_SELECTION: 'multiple_selection',
  TRUE_FALSE: 'true_false',
  INPUT_QUESTION: 'input_question',
};

export const AssessmentItemTypeLabels = {
  [AssessmentItemTypes.SINGLE_SELECTION]: 'Single selection',
  [AssessmentItemTypes.MULTIPLE_SELECTION]: 'Multiple selection',
  [AssessmentItemTypes.TRUE_FALSE]: 'True/False',
  [AssessmentItemTypes.INPUT_QUESTION]: 'Input question',
};
