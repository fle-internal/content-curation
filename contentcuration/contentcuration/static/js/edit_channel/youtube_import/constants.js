export const PageTypes = {
  SUBMIT_URL: 'SUBMIT_URL',
  CONFIRM: 'CONFIRM',
};

export const ImportStatus = {
  IDLE: 'IDLE',
  EXTRACTING: 'EXTRACTING',
  EXTRACT_ERROR: 'EXTRACT_ERROR',
  DOWNLOADING: 'DOWNLOADING',
  DOWNLOAD_ERROR: 'DOWNLOAD_ERROR',
  START_IMPORT: 'START_IMPORT',
  DONE: 'DONE',
};
