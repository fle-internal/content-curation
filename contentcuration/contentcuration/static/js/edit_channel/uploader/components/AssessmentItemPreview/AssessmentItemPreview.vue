<template>
  <div>
    <VLayout align-top justify-space-between>
      <VFlex>
        <div class="caption grey--text mb-1">
          {{ kindLabel }}
        </div>
        <div>
          {{ question }}
        </div>
      </VFlex>
    </VLayout>

    <VLayout v-if="detailed" mt-3>
      <VFlex>
        <div class="caption grey--text mb-2">
          Answers
        </div>

        <div v-if="!answers || !answers.length">
          No answers yet
        </div>

        <template v-else>
          <template v-if="isSingleSelection || isTrueFalse">
            <VRadioGroup v-model="correctAnswersIndices">
              <VRadio
                v-for="(answer, idx) in answers"
                :key="idx"
                :label="answer.answer"
                :value="idx"
                readonly
              />
            </VRadioGroup>
          </template>

          <template v-if="isMultipleSelection">
            <VCheckbox
              v-for="(answer, idx) in answers"
              :key="idx"
              v-model="correctAnswersIndices"
              :label="answer.answer"
              :value="idx"
              readonly
            />
          </template>

          <VList v-if="isInputQuestion">
            <VListTile
              v-for="(answer, idx) in answers"
              :key="idx"
            >
              {{ answer.answer }}
            </VListTile>
          </VList>
        </template>

        <div class="mt-1">
          <!--
            class="hints-preview" is needed for precise click
            target detection in AssessmentView.vue
          -->
          <span v-if="hintsCount === 0">No hints yet</span>
          <div v-else class="hints-preview">
            <span
              class="hints-toggle"
              data-test="hintsToggle"
              @click="areHintsOpen= !areHintsOpen"
            >
              <span>{{ hintsToggleLabel }}</span>

              <span class="icon">
                <v-icon v-if="areHintsOpen">arrow_drop_down</v-icon>
                <v-icon v-else>arrow_drop_up</v-icon>
              </span>
            </span>

            <VList v-if="areHintsOpen">
              <VListTile
                v-for="(hint, hintIdx) in hints"
                :key="hintIdx"
              >
                <VFlex xs1>
                  {{ hint.order }}
                </VFlex>
                <VFlex>
                  {{ hint.hint }}
                </VFlex>
              </VListTile>
            </VList>
          </div>
        </div>
      </VFlex>
    </VLayout>
  </div>
</template>

<script>

  import { AssessmentItemTypes, AssessmentItemTypeLabels } from '../../constants';
  import { getCorrectAnswersIndices } from '../../utils';

  export default {
    name: 'AssessmentItemPreview',
    props: {
      question: {
        type: String,
      },
      kind: {
        type: String,
        validator: value => {
          return Object.values(AssessmentItemTypes).includes(value);
        },
      },
      answers: {
        type: Array,
      },
      hints: {
        type: Array,
      },
      detailed: {
        type: Boolean,
        default: false,
      },
    },
    data() {
      return {
        areHintsOpen: false,
      };
    },
    computed: {
      kindLabel() {
        return AssessmentItemTypeLabels[this.kind];
      },
      isSingleSelection() {
        return this.kind === AssessmentItemTypes.SINGLE_SELECTION;
      },
      isMultipleSelection() {
        return this.kind === AssessmentItemTypes.MULTIPLE_SELECTION;
      },
      isTrueFalse() {
        return this.kind === AssessmentItemTypes.TRUE_FALSE;
      },
      isInputQuestion() {
        return this.kind === AssessmentItemTypes.INPUT_QUESTION;
      },
      correctAnswersIndices() {
        return getCorrectAnswersIndices(this.kind, this.answers);
      },
      hintsToggleLabel() {
        if (this.hintsCount === 0) {
          return 'No hints yet';
        }

        if (this.areHintsOpen) {
          return 'Hide hints';
        }

        if (this.hintsCount === 1) {
          return 'Show 1 hint';
        }

        return `Show ${this.hintsCount} hints`;
      },
      hintsCount() {
        if (!this.hints) {
          return 0;
        }

        return this.hints.length;
      },
    },
  };

</script>

<style lang="less" scoped>

  .hints-toggle {
    text-decoration: underline;
    cursor: pointer;

    .icon {
      position: relative;
      top: 1px;
    }
  }

  /deep/ .v-input--selection-controls {
    margin-top: 0;

    .accent--text {
      color: gray !important;
    }

    .v-input__slot {
      margin-bottom: 0 !important;
    }
  }

</style>
