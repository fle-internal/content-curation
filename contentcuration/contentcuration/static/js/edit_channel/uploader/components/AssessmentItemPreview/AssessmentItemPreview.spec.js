import { shallowMount, mount } from '@vue/test-utils';

import { AssessmentItemTypes } from '../../constants';
import AssessmentItemPreview from './AssessmentItemPreview';

describe('AssessmentItemPreview', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(AssessmentItemPreview, {
      propsData: {
        question: 'Question',
        kind: AssessmentItemTypes.SINGLE_SELECTION,
        answers: [
          { answer: 'Answer 1', correct: false, order: 1 },
          { answer: 'Answer 2', correct: true, order: 2 },
          { answer: 'Answer 3', correct: false, order: 3 },
        ],
        hints: [{ hint: 'Hint 1', order: 1 }, { hint: 'Hint 2', order: 2 }],
      },
    });
  });

  it('smoke test', () => {
    const wrapper = shallowMount(AssessmentItemPreview);

    expect(wrapper.isVueInstance()).toBe(true);
  });

  it('renders', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('if detailed true', () => {
    beforeEach(() => {
      wrapper.setProps({
        detailed: true,
      });
    });

    it('displays answers and hints toggle', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('displays hints on hints toggle click', () => {
      wrapper.find('[data-test="hintsToggle"]').trigger('click');

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
