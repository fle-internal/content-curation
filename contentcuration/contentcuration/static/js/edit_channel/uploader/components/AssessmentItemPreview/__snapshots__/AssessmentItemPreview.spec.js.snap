// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentItemPreview if detailed true displays answers and hints toggle 1`] = `
<div>
  <div class="layout align-top justify-space-between">
    <div class="flex">
      <div class="caption grey--text mb-1">
        Single selection
      </div>
      <div>
        Question
      </div>
    </div>
  </div>
  <div class="layout mt-3">
    <div class="flex">
      <div class="caption grey--text mb-2">
        Answers
      </div>
      <div class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light">
        <div class="v-input__control">
          <div class="v-input__slot" style="height: auto;">
            <div role="radiogroup" class="v-input--radio-group__input">
              <div class="v-radio theme--light">
                <div class="v-input--selection-controls__input"><input aria-label="Answer 1" aria-checked="false" role="radio" type="radio" name="v-radio-9" value="0">
                  <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">radio_button_unchecked</i>
                </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">Answer 1</label>
              </div>
              <div class="v-radio theme--light accent--text">
                <div class="v-input--selection-controls__input"><input aria-label="Answer 2" aria-checked="true" role="radio" type="radio" name="v-radio-9" value="1">
                  <div class="v-input--selection-controls__ripple accent--text"></div><i aria-hidden="true" class="v-icon material-icons theme--light accent--text">radio_button_checked</i>
                </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">Answer 2</label>
              </div>
              <div class="v-radio theme--light">
                <div class="v-input--selection-controls__input"><input aria-label="Answer 3" aria-checked="false" role="radio" type="radio" name="v-radio-9" value="2">
                  <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">radio_button_unchecked</i>
                </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">Answer 3</label>
              </div>
            </div>
          </div>
          <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
        </div>
      </div>
      <!---->
      <!---->
      <div class="mt-1">
        <div class="hints-preview"><span data-test="hintsToggle" class="hints-toggle"><span>Show 2 hints</span> <span class="icon"><i aria-hidden="true" class="v-icon material-icons theme--light">arrow_drop_up</i></span></span>
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AssessmentItemPreview if detailed true displays hints on hints toggle click 1`] = `
<div>
  <div class="layout align-top justify-space-between">
    <div class="flex">
      <div class="caption grey--text mb-1">
        Single selection
      </div>
      <div>
        Question
      </div>
    </div>
  </div>
  <div class="layout mt-3">
    <div class="flex">
      <div class="caption grey--text mb-2">
        Answers
      </div>
      <div class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light">
        <div class="v-input__control">
          <div class="v-input__slot" style="height: auto;">
            <div role="radiogroup" class="v-input--radio-group__input">
              <div class="v-radio theme--light">
                <div class="v-input--selection-controls__input"><input aria-label="Answer 1" aria-checked="false" role="radio" type="radio" name="v-radio-21" value="0">
                  <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">radio_button_unchecked</i>
                </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">Answer 1</label>
              </div>
              <div class="v-radio theme--light accent--text">
                <div class="v-input--selection-controls__input"><input aria-label="Answer 2" aria-checked="true" role="radio" type="radio" name="v-radio-21" value="1">
                  <div class="v-input--selection-controls__ripple accent--text"></div><i aria-hidden="true" class="v-icon material-icons theme--light accent--text">radio_button_checked</i>
                </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">Answer 2</label>
              </div>
              <div class="v-radio theme--light">
                <div class="v-input--selection-controls__input"><input aria-label="Answer 3" aria-checked="false" role="radio" type="radio" name="v-radio-21" value="2">
                  <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">radio_button_unchecked</i>
                </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">Answer 3</label>
              </div>
            </div>
          </div>
          <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
        </div>
      </div>
      <!---->
      <!---->
      <div class="mt-1">
        <div class="hints-preview"><span data-test="hintsToggle" class="hints-toggle"><span>Hide hints</span> <span class="icon"><i aria-hidden="true" class="v-icon material-icons theme--light">arrow_drop_down</i></span></span>
          <div role="list" class="v-list theme--light">
            <div role="listitem">
              <div class="v-list__tile theme--light">
                <div class="flex xs1">
                  1
                </div>
                <div class="flex">
                  Hint 1
                </div>
              </div>
            </div>
            <div role="listitem">
              <div class="v-list__tile theme--light">
                <div class="flex xs1">
                  2
                </div>
                <div class="flex">
                  Hint 2
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AssessmentItemPreview renders 1`] = `
<div>
  <div class="layout align-top justify-space-between">
    <div class="flex">
      <div class="caption grey--text mb-1">
        Single selection
      </div>
      <div>
        Question
      </div>
    </div>
  </div>
  <!---->
</div>
`;
